File: app/src/main/java/com/example/pokeapp/utilities/ScreenDimensions.java
package com.example.pokeapp.utilities;

import android.content.Context;


/**
 * This class provides utilities methods to get screen dimensions.
 */
public class ScreenDimensions {
    /**
     * Determines screen width in pixels.
     *
     * @param context Application context.
     * @return Screen width in pixels.
     */
    public static int getWidthPixels(Context context) {
        return context.getResources().getDisplayMetrics().widthPixels;
    }

    /**
     * Determines screen height in pixels.
     *
     * @param context Application context.
     * @return Screen height in pixels.
     */
    public static int getHeightPixels(Context context) {
        return context.getResources().getDisplayMetrics().heightPixels;
    }

    /**
     * Determines the density of the display.
     *
     * @param context Application context.
     * @return Density of the display.
     */
    public static float getDisplayDensity(Context context) {
        return context.getResources().getDisplayMetrics().density;
    }

    /**
     * Indicates if the screen has at least 240px in the Y axis.
     *
     * @param context Application context.
     * @return True if the screen has at least 240px in the Y axis, false otherwise.
     */
    public static boolean screenHasAtLeast240pxInYAxis(Context context) {
        return context.getResources().getDisplayMetrics().ydpi >= 240;
    }
}

File: app/src/main/java/com/example/pokeapp/utilities/ScreenDimensions.kt
package com.example.pokeapp.utilities

import android.content.Context

/**
 * This class provides utilities methods to get screen dimensions.
 */
object ScreenDimensions {
    /**
     * Determines screen width in pixels.
     *
     * @param context Application context.
     * @return Screen width in pixels.
     */
    fun getWidthPixels(context: Context): Int {
        return context.resources.displayMetrics.widthPixels
    }

    /**
     * Determines screen height in pixels.
     *
     * @param context Application context.
     * @return Screen height in pixels.
     */
    fun getHeightPixels(context: Context): Int {
        return context.resources.displayMetrics.heightPixels
    }

    /**
     * Determines the density of the display.
     *
     * @param context Application context.
     * @return Density of the display.
     */
    fun getDisplayDensity(context: Context): Float {
        return context.resources.displayMetrics.density
    }

    /**
     * Indicates if the screen has at least 240px in the Y axis.
     *
     * @param context Application context.
     * @return True if the screen has at least 240px in the Y axis, false otherwise.
     */
    fun screenHasAtLeast240pxInYAxis(context: Context): Boolean {
        return context.resources.displayMetrics.ydpi >= 240
    }
}